src/stratigraphy/util/depthcolumn.py

```diff
@@ -262,14 +262,14 @@ def depth_intervals(self) -> list[BoundaryInterval]:
         return depth_intervals
 
     def significant_arithmetic_progression(self) -> bool:
-        if len(self.entries) < 6:
+        if len(self.entries) < 7:
```

Comment:
This simply improved the scores.

Reply:
Keeping the segment length at 6 and simplifying the threshold to `abs(scale_pearson_correlation_coef) >= 0.9999` regardless of segment length seems to work even better. (Implemented in my commit.)